pagopa/io-app-design-system
# [IOAPPX-337] Refactor `Modules…` and `NumberPad` to take advantage of the new `HStack` and `VStack` components

>[!caution]
> This PR depends on the following PR. **Do not merge the PR beforehand, otherwise it might break part of the UI**.
> * https://github.com/pagopa/io-app/pull/5864

## Short description
This PR refactors `Modules…` and `NumberPad` components to take advantage of the new `HStack` and `VStack` components.

## List of changes proposed in this pull request
- Update `NumberPad` and `CodeInput` to replace the logic using `VSpacer` and `HSpacer`
- Update all the `Modules…` components (both interactive and skeleton states) using `HStack` and `VStack`
- Refactor `HStack` and `VStack` to apply some specific style properties (`alignItems`, `flexShrink`, etc…)

### Related PRs
* https://github.com/pagopa/io-app-design-system/pull/241
* https://github.com/pagopa/io-app-design-system/pull/297

## How to test
1. Launch the example app
2. First, check **Numberpad** screen
3. Second, check **Modules** screen
